### DISCUSSION TITLE
How to obtain the "convolved data" array?

### DISCUSSION THREAD

**Discussion by cosmic-nori** [sep-developers/sep#129] [Q&A] [answered]
I want to check the numpy array of "convolved data" which is derived by `sep.extract` by setting `filter_kernel` parameter.
Is there any way to obtain the "convolved data" array? Thanks.

**Comment by kbarbary** [comment, marked as answer]
There isn't a way to obtain the convolved data array, because it is calculated "on the fly" one line at a time: the entire array is not stored in memory at any point. However, it should be the same as what you would get using [`scipy.signal.convolve(data, filter_kernel, mode="same")`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.convolve.html). (It may differ near the image boundaries a bit, not entirely sure).

**Reply by cosmic-nori** [reply]
Thank you for your answer. I tried to use `scipy.signal.convolve` to reproduce the convolved data array which is calculated in SEP and I found it looks fairly consistent with SEP results. (I checked the number of detected objects with the same thresholding parameters and the results differ only a little bit.)
Then I want to ask a question. In SEP, a `filter_kernel` is normalized by the maximum value of the array, right? (e.g., `filter_kernel` / np.max(`filter_kernel`))

**Comment by kbarbary** [comment]
It's normalized by the sum of the array, so that the normalized kernel sums to 1. The normalization is done in [these lines](https://github.com/kbarbary/sep/blob/main/src/extract.c#L372-L378).

**Reply by cosmic-nori** [reply]
I see, got it. Thanks!
